engine/execution/computation/computer/computer.go

```diff
@@ -219,9 +226,46 @@ func (e *blockComputer) executeBlock(
 
 	res.StateReads = stateView.(*delta.View).ReadsCount()
 
+	executionData := generateExecutionData(res, collections, systemCol)
+
+	executionDataID, err := e.executionDataProvider.Provide(ctx, block.Height(), executionData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to provide execution data: %w", err)
+	}
+
+	res.ExecutionDataID = executionDataID
+
 	return res, nil
 }
 
+func generateExecutionData(
+	res *execution.ComputationResult,
+	collections []*entity.CompleteCollection,
+	systemCol *flow.Collection,
+) *execution_data.BlockExecutionData {
+	executionData := &execution_data.BlockExecutionData{
+		BlockID:             res.ExecutableBlock.ID(),
+		ChunkExecutionDatas: make([]*execution_data.ChunkExecutionData, 0, len(collections)+1),
+	}
+
+	for i, collection := range collections {
+		col := collection.Collection()
+		executionData.ChunkExecutionDatas = append(executionData.ChunkExecutionDatas, &execution_data.ChunkExecutionData{
+			Collection: &col,
+			Events:     res.Events[i],
```

Comment:
⚠️ I don't think this is correct, a collection might contain multiple events.

Reply:
It's corrext, `res.Events` is a 2d array